public/pages/Login.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from 'preact/hooks';
+
+import { useRootState } from '../store';
+import { login } from '../store/actions';
+
+export default function Register() {
+    const formRef = createRef<HTMLFormElement>();
```

Comment:
This is quite bad, if you did not know. 

Only use `createRef()` in class based components. `createRef()` will always create a new reference, and this is fine in class-based components where you have a constructor. In functional components though, this means you are creating a new reference on every single render. This is where `useRef()` comes in. It'll return the same ref upon every render.

The names are super similar so might've just been a typo in the suggestion from your IDE for all I know. Just thought I'd mention it.

Reply:
You are right. It's a mistake.